mmaction/datasets/pipelines/pose_loading.py

```diff
@@ -0,0 +1,666 @@
+        results['clip_len'] = self.clip_len
+        results['frame_interval'] = None
+        results['num_clips'] = self.num_clips
+        return results
+
+    def __repr__(self):
+        repr_str = (f'{self.__class__.__name__}('
+                    f'clip_len={self.clip_len}, '
+                    f'num_clips={self.num_clips}, '
+                    f'test_mode={self.test_mode}, '
+                    f'seed={self.seed})')
+        return repr_str
+
+
+@PIPELINES.register_module()
+class PoseDecode(object):
+    """Load and decode pose with given indices.
+
+    Required keys are "kp", "frame_inds" (optional), "kpscore" (optional),
+    added or modified keys are "kp", "kpscore" (if applicable).
+
+    Args:
+        random_drop (bool): Whether to randomly drop keypoints. The following
+            args are applicable only when `random_crop == True`. When set as
+            True, "kpscore" is a mandatory key. Default: False.
+        random_seed (int): Random seed used for randomly dropping keypoints.
+            Default: 1.
+        drop_prob (float): The probability for dropping one keypoint for each
+            frame. Default: 1 / 16.
+        manipulate_joints (tuple[int]): The joint indexes that may be dropped.
+            Default: (7, 8, 9, 10, 13, 14, 15, 16). (limb joints)
+    """
+
+    def __init__(self,
+                 random_drop=False,
+                 random_seed=1,
+                 drop_prob=1. / 16.,
+                 manipulate_joints=(7, 8, 9, 10, 13, 14, 15, 16)):
+        self.random_drop = random_drop
+        self.random_seed = random_seed
+        self.drop_prob = drop_prob
+        self.manipulate_joints = manipulate_joints
+
+    # inplace
+    def _drop_kpscore(self, kpscores):
+        """Randomly drop keypoints by setting the corresponding keypoint scores
+        as 0.
+
+        Args:
+            kpscores (np.ndarray): The confidence scores of keypoints.
+        """
+
+        for kpscore in kpscores:
+            lt = kpscore.shape[0]
+            for tidx in range(lt):
+                if np.random.random() < self.drop_prob:
+                    jidx = np.random.choice(self.manipulate_joints)
+                    kpscore[tidx, jidx] = 0.
+
+    def _load_kp(self, kp, frame_inds):
+        """Load keypoints given frame indices.
+
+        Args:
+            kp (np.ndarray): The keypoint coordinates.
+            frame_inds (np.ndarray): The frame indices.
+        """
+
+        return [x[frame_inds].astype(np.float32) for x in kp]
+
+    def _load_kpscore(self, kpscore, frame_inds):
+        """Load keypoint scores given frame indices.
+
+        Args:
+            kpscore (np.ndarray): The confidence scores of keypoints.
+            frame_inds (np.ndarray): The frame indices.
+        """
+
+        return [x[frame_inds].astype(np.float32) for x in kpscore]
+
+    def __call__(self, results):
+        if self.random_drop:
+            np.random.seed(self.random_seed)
```

Comment:
Why set seed

Reply:
For the same reason: we want the testing to be deterministic.
Actually, these options are used by the ablation experiments, shall I remove it?

Reply:
Yes, removing is a better option